tinkerpop3/src/main/java/io/shiftleft/overflowdb/NodeLayoutInformation.java

```diff
@@ -51,19 +54,25 @@ public NodeLayoutInformation(Set<String> propertyKeys,
     /* create unique offsets for each edge type and direction
      * sort them by edge label to ensure we get the same offsets between restarts
      * n.b. this doesn't support schema changes */
+
+    allowedOutEdgeLabels = new String[outEdgeLayouts.size()];
+    allowedInEdgeLabels = new String[inEdgeLayouts.size()];
     int offsetPosition = 0;
+    nPropertiesbyOffsetPos = new int[outEdgeLayouts.size() + inEdgeLayouts.size()];
+    int i = 0;
     outEdgeToOffsetPosition = new HashMap<>(outEdgeLayouts.size());
-    for (EdgeLayoutInformation edgeLayout : sortByLabel(outEdgeLayouts)) {
```

Comment:
Sorting the layouts ensures that we get consistent layouts in between jvm starts. 
If you stop sorting it by label we get a different layout every time we start, i.e. we cannot save to disk and load it back in a new jvm instance.

Reply:
The layouts are passed in a List, which is not a set/hashmap. The new rule is: We use the same ordering as the caller in the constructor. Is that acceptable?

Reply:
true, looking at the implementation now, seems ok to not sort again.